code/mob/living/critter/flock/flockdrone.dm

```diff
@@ -775,6 +789,13 @@
 	remove_lifeprocess(/datum/lifeprocess/statusupdate)
 
 /mob/living/critter/flock/drone/death(var/gibbed)
+	if (src.selected_by)
+		var/mob/living/intangible/flock/selector = src.selected_by
+		var/datum/abilityHolder/flockmind/AH = selector.abilityHolder
+		AH.drone_controller.cast(src)
+		selector.targeting_ability = null
+		selector.update_cursor()
```

Comment:
this block should probably be a proc since it's duplicated several times (+- the `update_cursor` call, which I assume is always meant to be there?)

Reply:
update_cursor() wasn't needed in a few cases, not sure why